resources/providers/azurelib/auth/credentials_test.go

```diff
@@ -44,13 +44,26 @@ func TestConfigProvider_GetAzureClientConfig(t *testing.T) {
 				Credentials: &azidentity.DefaultAzureCredential{},
 			},
 		},
+		{
+			name: "Should return a DefaultAzureCredential on unknown client credentials type",
+			config: config.AzureConfig{
+				Credentials: config.AzureClientOpt{
+					ClientCredentialsType: "unknown",
+				},
+			},
+			authProviderInitFn: initDefaultCredentialsMock(nil),
+			want: &AzureFactoryConfig{
```

Comment:
Sorry for being nitpicky but this should probably return an error, it signifies a mismatch between integration and cloubeat

Reply:
Sure,  I added an error in that case. Not in an empty string, though. Empty string fallbacks to default.